editoast/fga/src/lib.rs

```diff
@@ -516,16 +516,50 @@ pub mod test_utilities {
     macro_rules! test_client {
         () => {
             $crate::client::Client::try_new_store(
-                stdext::function_name!()
-                    .split("::")
-                    .filter(|x| *x != "{{closure}}")
-                    .collect::<Vec<_>>()
-                    .join("-"),
+                $crate::test_utilities::sanitize_store_name(
+                    &stdext::function_name!()
+                        .split("::")
+                        .filter(|x| *x != "{{closure}}")
+                        .collect::<Vec<_>>()
+                        .join("-"),
+                ),
                 $crate::test_utilities::connection_settings(),
             )
             .await
             .expect("Failed to initialize client")
         };
+        ($store_suffix:expr) => {
+            $crate::client::Client::try_new_store(
+                $crate::test_utilities::sanitize_store_name(&format!(
+                    "{}@{}",
+                    $store_suffix.to_string(),
+                    stdext::function_name!()
+                        .split("::")
+                        .filter(|x| *x != "{{closure}}")
+                        .collect::<Vec<_>>()
+                        .join("-")
+                )),
+                $crate::test_utilities::connection_settings(),
+            )
+            .await
+            .expect("Failed to initialize client")
+        };
+    }
+
+    pub fn sanitize_store_name(store_name: &str) -> String {
```

Comment:
```suggestion
    pub fn sanitize_store_name_length(store_name: &str) -> String {
```

You could still theoretically send the right amount of wrong characters.